### ISSUE TITLE
IllegalArgumentException: requirement failed

### ISSUE THREAD

**Issue by RadoBuransky** [JetBrains/intellij-scala#161]
Hi. I just got this error. It happened only once. It's not a blocking issue.

IntelliJ IDEA 14.1.3 #IC-141.1010
Scala Plugin 1.5.2

```
requirement failed
java.lang.IllegalArgumentException: requirement failed
    at scala.Predef$.require(Predef.scala:207)
    at org.jetbrains.plugins.scala.lang.psi.impl.toplevel.typedef.TypeDefinitionMembers$Lazy.apply(TypeDefinitionMembers.scala:706)
    at org.jetbrains.plugins.scala.lang.psi.impl.toplevel.typedef.TypeDefinitionMembers$.privateProcessDeclarations(TypeDefinitionMembers.scala:923)
    at org.jetbrains.plugins.scala.lang.psi.impl.toplevel.typedef.TypeDefinitionMembers$.processDeclarations(TypeDefinitionMembers.scala:653)
    at org.jetbrains.plugins.scala.lang.psi.api.toplevel.typedef.ScTemplateDefinition$class.processDeclarationsForTemplateBody(ScTemplateDefinition.scala:326)
    at org.jetbrains.plugins.scala.lang.psi.impl.toplevel.typedef.ScObjectImpl.processDeclarationsForTemplateBody(ScObjectImpl.scala:105)
    at org.jetbrains.plugins.scala.lang.psi.api.toplevel.typedef.ScTemplateDefinition$class.processDeclarations(ScTemplateDefinition.scala:260)
    at org.jetbrains.plugins.scala.lang.psi.impl.toplevel.typedef.ScObjectImpl.processDeclarations(ScObjectImpl.scala:123)
    at org.jetbrains.plugins.scala.lang.psi.impl.toplevel.packaging.ScPackagingImpl.processDeclarations(ScPackagingImpl.scala:144)
    at org.jetbrains.plugins.scala.lang.resolve.ResolvableStableCodeReferenceElement$class.treeWalkUp$1(ResolvableStableCodeReferenceElement.scala:152)
    at org.jetbrains.plugins.scala.lang.resolve.ResolvableStableCodeReferenceElement$class.processQualifier(ResolvableStableCodeReferenceElement.scala:162)
    at org.jetbrains.plugins.scala.lang.psi.impl.base.ScStableCodeReferenceElementImpl.processQualifier(ScStableCodeReferenceElementImpl.scala:33)
    at org.jetbrains.plugins.scala.lang.resolve.ResolvableStableCodeReferenceElement$class.doResolve(ResolvableStableCodeReferenceElement.scala:132)
    at org.jetbrains.plugins.scala.lang.psi.impl.base.ScStableCodeReferenceElementImpl.doResolve(ScStableCodeReferenceElementImpl.scala:33)
    at org.jetbrains.plugins.scala.lang.resolve.StableCodeReferenceElementResolver.resolve(StableCodeReferenceElementResolver.scala:40)
    at org.jetbrains.plugins.scala.lang.resolve.ResolvableStableCodeReferenceElement$$anonfun$multiResolve$2.apply(ResolvableStableCodeReferenceElement.scala:36)
    at org.jetbrains.plugins.scala.lang.resolve.ResolvableStableCodeReferenceElement$$anonfun$multiResolve$2.apply(ResolvableStableCodeReferenceElement.scala:36)
    at org.jetbrains.plugins.scala.caches.CachesUtil$$anon$4.compute(CachesUtil.scala:245)
    at com.intellij.openapi.util.RecursionManager$2.doPreventingRecursion(RecursionManager.java:112)
    at org.jetbrains.plugins.scala.caches.CachesUtil$.getMappedWithRecursionPreventingWithRollback(CachesUtil.scala:242)
    at org.jetbrains.plugins.scala.lang.resolve.ResolvableStableCodeReferenceElement$class.multiResolve(ResolvableStableCodeReferenceElement.scala:35)
    at org.jetbrains.plugins.scala.lang.psi.impl.base.ScStableCodeReferenceElementImpl.multiResolve(ScStableCodeReferenceElementImpl.scala:33)
    at org.jetbrains.plugins.scala.lang.resolve.ResolvableReferenceElement$class.bind(ResolvableReferenceElement.scala:28)
    at org.jetbrains.plugins.scala.lang.psi.impl.base.ScStableCodeReferenceElementImpl.bind(ScStableCodeReferenceElementImpl.scala:33)
    at org.jetbrains.plugins.scala.lang.resolve.ResolvableReferenceElement$class.resolve(ResolvableReferenceElement.scala:12)
    at org.jetbrains.plugins.scala.lang.psi.impl.base.ScStableCodeReferenceElementImpl.resolve(ScStableCodeReferenceElementImpl.scala:33)
    at org.jetbrains.plugins.scala.lang.psi.impl.base.types.ScSimpleTypeElementImpl$.calculateReferenceType(ScSimpleTypeElementImpl.scala:398)
    at org.jetbrains.plugins.scala.lang.psi.impl.base.types.ScSimpleTypeElementImpl$.calculateReferenceType(ScSimpleTypeElementImpl.scala:408)
    at org.jetbrains.plugins.scala.lang.psi.impl.base.types.ScSimpleTypeElementImpl.org$jetbrains$plugins$scala$lang$psi$impl$base$types$ScSimpleTypeElementImpl$$innerNonValueType(ScSimpleTypeElementImpl.scala:335)
    at org.jetbrains.plugins.scala.lang.psi.impl.base.types.ScSimpleTypeElementImpl.innerType(ScSimpleTypeElementImpl.scala:56)
    at org.jetbrains.plugins.scala.lang.psi.api.base.types.ScTypeElement$$anonfun$getType$1$$anonfun$apply$1.apply(ScTypeElement.scala:22)
    at org.jetbrains.plugins.scala.lang.psi.api.base.types.ScTypeElement$$anonfun$getType$1$$anonfun$apply$1.apply(ScTypeElement.scala:22)
    at org.jetbrains.plugins.scala.caches.CachesUtil$MyProvider.compute(CachesUtil.scala:183)
    at org.jetbrains.plugins.scala.caches.CachesUtil$$anon$1$$anon$2.compute(CachesUtil.scala:134)
    at org.jetbrains.plugins.scala.caches.CachesUtil$$anon$1$$anon$2.compute(CachesUtil.scala:131)
    at com.intellij.openapi.util.RecursionManager$2.doPreventingRecursion(RecursionManager.java:112)
    at org.jetbrains.plugins.scala.caches.CachesUtil$$anon$1.compute(CachesUtil.scala:131)
    at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:49)
    at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:200)
    at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:38)
    at org.jetbrains.plugins.scala.caches.CachesUtil$.getWithRecursionPreventingWithRollback(CachesUtil.scala:163)
    at org.jetbrains.plugins.scala.lang.psi.api.base.types.ScTypeElement$class.getType(ScTypeElement.scala:20)
    at org.jetbrains.plugins.scala.lang.psi.impl.base.types.ScSimpleTypeElementImpl.getType(ScSimpleTypeElementImpl.scala:39)
    at org.jetbrains.plugins.scala.lang.psi.impl.base.types.ScParameterizedTypeElementImpl$$anonfun$4.apply(ScParameterizedTypeElementImpl.scala:117)
    at org.jetbrains.plugins.scala.lang.psi.impl.base.types.ScParameterizedTypeElementImpl$$anonfun$4.apply(ScParameterizedTypeElementImpl.scala:117)
    at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245)
    at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245)
    at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
    at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:35)
    at scala.collection.TraversableLike$class.map(TraversableLike.scala:245)
    at scala.collection.AbstractTraversable.map(Traversable.scala:104)
    at org.jetbrains.plugins.scala.lang.psi.impl.base.types.ScParameterizedTypeElementImpl.innerType(ScParameterizedTypeElementImpl.scala:117)
    at org.jetbrains.plugins.scala.lang.psi.api.base.types.ScTypeElement$$anonfun$getType$1$$anonfun$apply$1.apply(ScTypeElement.scala:22)
    at org.jetbrains.plugins.scala.lang.psi.api.base.types.ScTypeElement$$anonfun$getType$1$$anonfun$apply$1.apply(ScTypeElement.scala:22)
    at org.jetbrains.plugins.scala.caches.CachesUtil$MyProvider.compute(CachesUtil.scala:183)
    at org.jetbrains.plugins.scala.caches.CachesUtil$$anon$1$$anon$2.compute(CachesUtil.scala:134)
    at org.jetbrains.plugins.scala.caches.CachesUtil$$anon$1$$anon$2.compute(CachesUtil.scala:131)
    at com.intellij.openapi.util.RecursionManager$2.doPreventingRecursion(RecursionManager.java:112)
    at org.jetbrains.plugins.scala.caches.CachesUtil$$anon$1.compute(CachesUtil.scala:131)
    at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:49)
    at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:200)
    at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:38)
    at org.jetbrains.plugins.scala.caches.CachesUtil$.getWithRecursionPreventingWithRollback(CachesUtil.scala:163)
    at org.jetbrains.plugins.scala.lang.psi.api.base.types.ScTypeElement$class.getType(ScTypeElement.scala:20)
    at org.jetbrains.plugins.scala.lang.psi.impl.base.types.ScParameterizedTypeElementImpl.getType(ScParameterizedTypeElementImpl.scala:22)
    at org.jetbrains.plugins.scala.lang.psi.impl.statements.ScPatternDefinitionImpl.getType(ScPatternDefinitionImpl.scala:58)
    at org.jetbrains.plugins.scala.lang.psi.api.base.patterns.ScPattern$class.org$jetbrains$plugins$scala$lang$psi$api$base$patterns$ScPattern$$innerExpectedType(ScPattern.scala:236)
    at org.jetbrains.plugins.scala.lang.psi.api.base.patterns.ScPattern$$anonfun$expectedType$1$$anonfun$apply$1.apply(ScPattern.scala:225)
    at org.jetbrains.plugins.scala.lang.psi.api.base.patterns.ScPattern$$anonfun$expectedType$1$$anonfun$apply$1.apply(ScPattern.scala:225)
    at org.jetbrains.plugins.scala.caches.CachesUtil$MyProvider.compute(CachesUtil.scala:183)
    at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:49)
    at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:200)
    at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:38)
    at org.jetbrains.plugins.scala.caches.CachesUtil$.get(CachesUtil.scala:173)
    at org.jetbrains.plugins.scala.lang.psi.api.base.patterns.ScPattern$class.expectedType(ScPattern.scala:224)
    at org.jetbrains.plugins.scala.lang.psi.impl.base.patterns.ScReferencePatternImpl.expectedType(ScReferencePatternImpl.scala:27)
    at org.jetbrains.plugins.scala.lang.psi.impl.base.patterns.ScReferencePatternImpl.getType(ScReferencePatternImpl.scala:47)
    at org.jetbrains.plugins.scala.lang.psi.implicits.ScImplicitlyConvertible$CollectImplicitsProcessor.execute(ScImplicitlyConvertible.scala:436)
    at org.jetbrains.plugins.scala.lang.psi.impl.toplevel.typedef.TypeDefinitionMembers$.tail$1(TypeDefinitionMembers.scala:780)
    at org.jetbrains.plugins.scala.lang.psi.impl.toplevel.typedef.TypeDefinitionMembers$.runForValInfo$1(TypeDefinitionMembers.scala:776)
    at org.jetbrains.plugins.scala.lang.psi.impl.toplevel.typedef.TypeDefinitionMembers$.addSignature$3(TypeDefinitionMembers.scala:835)
    at org.jetbrains.plugins.scala.lang.psi.impl.toplevel.typedef.TypeDefinitionMembers$.process$1(TypeDefinitionMembers.scala:873)
    at org.jetbrains.plugins.scala.lang.psi.impl.toplevel.typedef.TypeDefinitionMembers$.privateProcessDeclarations(TypeDefinitionMembers.scala:925)
    at org.jetbrains.plugins.scala.lang.psi.impl.toplevel.typedef.TypeDefinitionMembers$.processDeclarations(TypeDefinitionMembers.scala:653)
    at org.jetbrains.plugins.scala.lang.psi.api.toplevel.typedef.ScTemplateDefinition$class.processDeclarationsForTemplateBody(ScTemplateDefinition.scala:326)
    at org.jetbrains.plugins.scala.lang.psi.impl.toplevel.typedef.ScObjectImpl.processDeclarationsForTemplateBody(ScObjectImpl.scala:105)
    at org.jetbrains.plugins.scala.lang.psi.api.toplevel.typedef.ScTemplateDefinition$class.processDeclarations(ScTemplateDefinition.scala:260)
    at org.jetbrains.plugins.scala.lang.psi.impl.toplevel.typedef.ScObjectImpl.processDeclarations(ScObjectImpl.scala:130)
    at org.jetbrains.plugins.scala.lang.psi.api.FileDeclarationsHolder$$anonfun$checkObjects$1$1.apply$mcV$sp(FileDeclarationsHolder.scala:136)
    at org.jetbrains.plugins.scala.lang.resolve.processor.BaseProcessor.definePriority(BaseProcessor.scala:62)
    at org.jetbrains.plugins.scala.lang.psi.api.FileDeclarationsHolder$class.updateProcessor(FileDeclarationsHolder.scala:197)
    at org.jetbrains.plugins.scala.lang.psi.api.FileDeclarationsHolder$class.checkObjects$1(FileDeclarationsHolder.scala:123)
    at org.jetbrains.plugins.scala.lang.psi.api.FileDeclarationsHolder$class.processDeclarations(FileDeclarationsHolder.scala:162)
    at org.jetbrains.plugins.scala.lang.psi.impl.ScalaFileImpl.processDeclarations(ScalaFileImpl.scala:44)
    at org.jetbrains.plugins.scala.lang.psi.implicits.ScImplicitlyConvertible.treeWalkUp$1(ScImplicitlyConvertible.scala:186)
    at org.jetbrains.plugins.scala.lang.psi.implicits.ScImplicitlyConvertible.org$jetbrains$plugins$scala$lang$psi$implicits$ScImplicitlyConvertible$$buildSimpleImplicitMapInner(ScImplicitlyConvertible.scala:195)
    at org.jetbrains.plugins.scala.lang.psi.implicits.ScImplicitlyConvertible$$anonfun$buildSimpleImplicitMap$2.apply(ScImplicitlyConvertible.scala:172)
    at org.jetbrains.plugins.scala.lang.psi.implicits.ScImplicitlyConvertible$$anonfun$buildSimpleImplicitMap$2.apply(ScImplicitlyConvertible.scala:172)
    at org.jetbrains.plugins.scala.caches.CachesUtil$$anon$4.compute(CachesUtil.scala:245)
    at com.intellij.openapi.util.RecursionManager$2.doPreventingRecursion(RecursionManager.java:112)
    at org.jetbrains.plugins.scala.caches.CachesUtil$.getMappedWithRecursionPreventingWithRollback(CachesUtil.scala:242)
    at org.jetbrains.plugins.scala.lang.psi.implicits.ScImplicitlyConvertible.buildSimpleImplicitMap(ScImplicitlyConvertible.scala:171)
    at org.jetbrains.plugins.scala.lang.psi.implicits.ScImplicitlyConvertible.org$jetbrains$plugins$scala$lang$psi$implicits$ScImplicitlyConvertible$$buildImplicitMap(ScImplicitlyConvertible.scala:116)
    at org.jetbrains.plugins.scala.lang.psi.implicits.ScImplicitlyConvertible$$anonfun$implicitMapFirstPart$2.apply(ScImplicitlyConvertible.scala:89)
    at org.jetbrains.plugins.scala.lang.psi.implicits.ScImplicitlyConvertible$$anonfun$implicitMapFirstPart$2.apply(ScImplicitlyConvertible.scala:89)
    at org.jetbrains.plugins.scala.caches.CachesUtil$$anon$4.compute(CachesUtil.scala:245)
    at com.intellij.openapi.util.RecursionManager$2.doPreventingRecursion(RecursionManager.java:112)
    at org.jetbrains.plugins.scala.caches.CachesUtil$.getMappedWithRecursionPreventingWithRollback(CachesUtil.scala:242)
    at org.jetbrains.plugins.scala.lang.psi.implicits.ScImplicitlyConvertible.implicitMapFirstPart(ScImplicitlyConvertible.scala:88)
    at org.jetbrains.plugins.scala.lang.psi.implicits.ScImplicitlyConvertible.implicitMap(ScImplicitlyConvertible.scala:67)
    at org.jetbrains.plugins.scala.lang.psi.api.expr.ScExpression$class.getImplicitConversions(ScExpression.scala:454)
    at org.jetbrains.plugins.scala.lang.psi.impl.expr.ScBlockExprImpl.getImplicitConversions(ScBlockExprImpl.scala:22)
    at org.jetbrains.plugins.scala.codeInsight.intention.expression.InlineImplicitConversionIntention.isAvailable(InlineImplicitConversionIntention.scala:32)
    at com.intellij.codeInsight.intention.PsiElementBaseIntentionAction.isAvailable(PsiElementBaseIntentionAction.java:64)
    at com.intellij.codeInsight.intention.impl.config.IntentionActionWrapper.isAvailable(IntentionActionWrapper.java:56)
    at com.intellij.codeInsight.intention.impl.ShowIntentionActionsHandler.isAvailableHere(ShowIntentionActionsHandler.java:125)
    at com.intellij.codeInsight.intention.impl.ShowIntentionActionsHandler.availableFor(ShowIntentionActionsHandler.java:106)
    at com.intellij.codeInsight.daemon.impl.ShowIntentionsPass$3.process(ShowIntentionsPass.java:297)
    at com.intellij.codeInsight.daemon.impl.ShowIntentionsPass$3.process(ShowIntentionsPass.java:294)
    at com.intellij.codeInsight.intention.impl.ShowIntentionActionsHandler.chooseBetweenHostAndInjected(ShowIntentionActionsHandler.java:150)
    at com.intellij.codeInsight.daemon.impl.ShowIntentionsPass.getActionsToShow(ShowIntentionsPass.java:293)
    at com.intellij.codeInsight.daemon.impl.ShowIntentionsPass.getIntentionActionsToShow(ShowIntentionsPass.java:212)
    at com.intellij.codeInsight.daemon.impl.ShowIntentionsPass.doCollectInformation(ShowIntentionsPass.java:188)
    at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:67)
    at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass$1$1.run(PassExecutorService.java:438)
    at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1098)
    at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass$1.run(PassExecutorService.java:429)
    at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:452)
    at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:402)
    at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
    at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:426)
    at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:402)
    at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask.exec(JobLauncherImpl.java:202)
    at jsr166e.ForkJoinTask.doExec(ForkJoinTask.java:260)
    at jsr166e.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:858)
    at jsr166e.ForkJoinPool.scan(ForkJoinPool.java:1687)
    at jsr166e.ForkJoinPool.runWorker(ForkJoinPool.java:1642)
    at jsr166e.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:109)
```

**Comment by Alefas** [comment]
We are using other issue tracker: https://youtrack.jetbrains.com/issues/SCL
For this particular issue we don't need any new issue.
This exception is very old, and I don't know how to fix it as there is no any clue in sources about what's wrong. I think it's some compiler problem, but I'm not ready yet to investigate bytecode as this exception is quite minor (doesn't affect anything).
